send/ftps_generic

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+import os
+import sys
+from ftplib import FTP_TLS
+from urllib.parse import urlparse
+
+import send_lib
+import tarfile
+
+service_name = "ftps_generic"
+port = 21
+
+if __name__ == "__main__":
+	# Parse command line arguments
+	send_lib.check_input_fields(sys.argv)
+	facility_name = sys.argv[1]
+	destination = sys.argv[2]
+
+	# Parse destination
+	# expected destination type url, example: 'ftp://example.org:2121/path-to/folder/'
+	parsed_destination = urlparse(destination, scheme='ftp')
+	if parsed_destination.scheme not in ['ftp', 'ftps', 'ftpes']:
+		print("only FTPS is supported by this service")
+		exit(1)
+	hostname = parsed_destination.hostname
+	port = parsed_destination.port or port
+	remote_folder_path = parsed_destination.path.rstrip('/')
+
+	local_file_path = os.getcwd() + "/../gen/spool/" + facility_name + "/" + service_name + "/" + service_name
+	remote_file_path = f'{remote_folder_path}/{service_name}'
+
+	# Connect via explicit FTPS (FTPES)
+	auth = send_lib.get_auth_credentials(service_name, destination)
+	ftps = FTP_TLS()
+	ftps.connect(hostname, port)
+	ftps.login(auth[0], auth[1])
+	ftps.prot_p()
+
+	# Send file to FTP server
+	with open(local_file_path, 'rb') as file:
+		ftps.storbinary(f'STOR {remote_file_path}', file)
```

Comment:
```suggestion
        remote_file_path = remote_file_path.replace('\\', '\\\\').replace('"', '\\"')
		ftps.storbinary(f'STOR "{remote_file_path}"', file)
```

if it can be assumed that no control characters (like newline) will be present in the destination string

Reply:
I added the replacement, but I removed the quotas around `"{remote_file_path}"`, without them it works correctly I think.

Reply:
The quotes are needed for it to be safe. If it does not work with the server, the validation needs to be changed to whitelisting a regex, something like `\w`